nilearn/plotting/surface/surf_plotting.py

```diff
@@ -0,0 +1,938 @@
+        passed and to 1 if a bg_map is passed.
+
+        .. note::
+            This option is currently only implemented for the
+            ``matplotlib`` engine.
+
+    %(vmin)s
+
+    %(vmax)s
+
+    %(symmetric_cbar)s
+
+    %(bg_on_data)s
+
+    %(darkness)s
+        Default=1.
+
+        .. note::
+            This option is currently only implemented for the
+            ``matplotlib`` engine.
+
+    %(title)s
+
+    title_font_size : :obj:`int`, default=None
+        Size of the title font (only implemented for the plotly engine).
+
+        .. versionadded:: 0.9.0
+
+    %(output_file)s
+
+    axes : instance of matplotlib axes or None, default=None
+        The axes instance to plot to. The projection must be '3d' (e.g.,
+        `figure, axes = plt.subplots(subplot_kw={'projection': '3d'})`,
+        where axes should be passed.).
+        If None, a new axes is created.
+
+        .. note::
+            This option is currently only implemented for the
+            ``matplotlib`` engine.
+
+    %(figure)s
+
+        .. note::
+            This option is currently only implemented for the
+            ``matplotlib`` engine.
+
+    %(avg_method)s
+
+        .. note::
+            This option is currently only implemented for the
+            ``matplotlib`` engine.
+
+        When using matplotlib as engine,
+        `avg_method` will default to ``"mean"`` if ``None`` is passed.
+
+        .. versionadded:: 0.10.3dev
+
+    kwargs : :obj:`dict`, optional
+        Keyword arguments passed to :func:`nilearn.plotting.plot_surf`.
+
+    See Also
+    --------
+    nilearn.datasets.fetch_surf_fsaverage: For surface data object to be
+        used as background map for this plotting function.
+
+    nilearn.plotting.plot_surf: For brain surface visualization.
+
+    nilearn.surface.vol_to_surf : For info on the generation of surfaces.
+    """
+    fig = get_surface_backend(engine).plot_surf_stat_map(
+        surf_mesh=surf_mesh,
+        stat_map=stat_map,
+        bg_map=bg_map,
+        hemi=hemi,
+        view=view,
+        threshold=threshold,
+        alpha=alpha,
+        vmin=vmin,
+        vmax=vmax,
+        cmap=cmap,
+        colorbar=colorbar,
+        symmetric_cbar=symmetric_cbar,
+        cbar_tick_format=cbar_tick_format,
+        bg_on_data=bg_on_data,
+        darkness=darkness,
+        title=title,
+        title_font_size=title_font_size,
+        output_file=output_file,
+        axes=axes,
+        figure=figure,
+        avg_method=avg_method,
+        **kwargs,
+    )
+    return fig
+
+
+@fill_doc
+def plot_img_on_surf(
+    stat_map,
+    surf_mesh="fsaverage5",
+    mask_img=None,
+    hemispheres=None,
+    bg_on_data=False,
+    inflate=False,
+    views=None,
+    output_file=None,
+    title=None,
+    colorbar=True,
+    vmin=None,
+    vmax=None,
+    threshold=None,
+    symmetric_cbar="auto",
+    cmap=DEFAULT_DIVERGING_CMAP,
+    cbar_tick_format="%i",
+    **kwargs,
+):
```

Comment:
This is not implemented for plotly. But I've put a method in plotly backend and raise NotImplementedError. I thought of adding `engine` parameter here. Would that be ok?